Documentation updated (#7)
fika-lang/fika: CONTRIBUTING.md

# Contributing

When contributing to this repository, please first discuss the change you wish to make via issue,

> email, or any other method with the owners of this repository before making a change.

Reviewer: How about we keep this to issue board and the discord for now. Issue board definitely is the right place to discuss the changes and bugs, but for more help and guidance for contribution we could use discord as well.

It gets easier for the maintainer to keep track of things

Author: Yeah sure, I'll add a discord link there so that users can directly join the discord group and ask whatever doubts they have in the channel itself ..

That's works right?

Reviewer: Yea, issue and discord works. We can remove "email, or any other method" part

Author: Ok, cool 👍

> Please note we have a code of conduct, please follow it in all your interactions with the project.

Reviewer: ```suggestion
Please make sure to follow our code of conduct in all your interactions with the project.
```

Author: Ok, I'll make the necessary changes in a while ...

**Thanks for reviewing the documentation**

If you don't have git on your machine, [install it](https://help.github.com/articles/set-up-git/).

## The beginner's guide to contributing to a GitHub project

<img align="right" width="300" src="assets/images/fork.png" alt="fork this repository" />

## Fork this repository

Fork this repository by clicking on the fork button on the top of this page.
This will create a copy of this repository in your account.

## Clone the repository

<img align="right" width="300" src="assets/images/clone.png" alt="clone this repository" />

Now clone the forked repository to your machine. Go to your GitHub account, open the forked repository, click on the code button and then click the _copy to clipboard_ icon.

Open a terminal and run the following git command:

```
git clone "url you just copied"
```

> where "url you just copied" (without the quotation marks) is the url to this repository (your fork of this project). See the previous steps to obtain the url.

Reviewer: Let's make it more obvious 😉 
you can put the actual repo url here. Easier for people to copy 😄 

And then you could remove the line which explains what the "url you just copied" is 🙂

Author: yeah, that makes sense 😄

<img align="right" width="300" src="assets/images/url.JPG" alt="copy URL to clipboard" />

For example:

```
git clone https://github.com/this-is-you/Fike.git
```

where `this-is-you` is your GitHub username. Here you're copying the contents of the first-contributions repository on GitHub to your computer.

## Create a branch

Change to the repository directory on your computer (if you are not already there):

```
cd Fika
```

Now create a branch using the `git checkout` command:

```
git checkout -b <add-your-new-branch-name>
```

For example:

```
git checkout -b add-alonzo-church
```

(The name of the branch does not need to have the word _add_ in it, but it's a reasonable thing to include because the purpose of this branch is to add your name to a list.)

## Make necessary changes and commit those changes

> Now open `Contributors.md` file in a text editor, add your name to it. Don't add it at the beginning or end of the file. Put it anywhere in between. Now, save the file.

Reviewer: @emilsoman are we keeping this? We don't have a Contributors.md file as of now
If we are keeping it, let's make one and add Emil and yourselves 🎉

Author: Ok i'll add Emil sir and me as contributor

Add those changes to the branch you just created using the `git add` command:

```
git add Contributors.md
```

Now commit those changes using the `git commit` command:

```
git commit -m "Add <your-name> to Contributors list"
```

replacing `<your-name>` with your name.

## Push changes to GitHub

Push your changes using the command `git push`:

```
git push origin <add-your-branch-name>
```

replacing `<add-your-branch-name>` with the name of the branch you created earlier.

## Submit your changes for review

If you go to your repository on GitHub, you'll see a `Compare & pull request` button. Click on that button.

<img style="float: right;" src="assets/images/compare-and-pull.png" alt="create a pull request" />

Now submit the pull request.

<img style="float: right;" src="assets/images/submit-pull-request.png" alt="submit pull request" />

Soon I'll be merging all your changes into the master branch of this project. You will get a notification email once the changes have been merged.

## Where to go from here?

Congrats! You just completed the standard _fork -> clone -> edit -> PR_ workflow that you'll encounter often as a contributor!

## Reference

> - [https://github.com/firstcontributions](https://github.com/firstcontributions)

Reviewer: Overall this a comprehensive flow documentation 👏

Author: Thanks for appreciating :)
